docs/x509.rst

```diff
@@ -447,6 +447,99 @@ X.509 Extensions
 
         Returns an instance of the extension type corresponding to the OID.
 
+.. class:: KeyUsage
+
+    .. versionadded:: 0.9
+
+    The key usage extension defines the purpose of the key contained in the
+    certificate.  The usage restriction might be employed when a key that could
+    be used for more than one operation is to be restricted. It corresponds to
+    :data:`OID_KEY_USAGE`.
+
+    .. attribute:: digital_signature
+
+        :type: bool
+
+        This is asserted when the subject public key is used for verifying
```

Comment:
Much better :-)

On Wed, Apr 1, 2015 at 11:26 PM, Paul Kehrer <notifications@github.com>
wrote:

> In docs/x509.rst
> <https://github.com/pyca/cryptography/pull/1802#discussion_r27630536>:
>
> > @@ -447,6 +447,99 @@ X.509 Extensions
> >
> >          Returns an instance of the extension type corresponding to the OID.
> >
> > +.. class:: KeyUsage
> > +
> > +    .. versionadded:: 0.9
> > +
> > +    The key usage extension defines the purpose of the key contained in the
> > +    certificate.  The usage restriction might be employed when a key that could
> > +    be used for more than one operation is to be restricted. It corresponds to
> > +    :data:`OID_KEY_USAGE`.
> > +
> > +    .. attribute:: digital_signature
> > +
> > +        :type: bool
> > +
> > +        This is asserted when the subject public key is used for verifying
>
> This was lifted pretty much straight from the RFC, so not surprising that
> it doesn't feel good [image: :grinning:]
>
> How about "This purpose is set to True when" or some variant of that?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1802/files#r27630536>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084